documentation/guides/zero-downtime-database-management.md

```diff
@@ -290,3 +290,149 @@ a
+different running versions of the application. For environment variables, the
+previously running version of the application will continue to use the
+environment variable configuration defined in the previous task definition
+revision used to run that instance. Environment variables are not automatically
+updated for the running application. This means that if a deployment was very
+carefully orchestrated, we do not necessarily need to worry about backwards
+incompatible changes, and even in the
+[worst case described below](#manual-rollbacks-after-removing-an-environment-variable),
+it is still possible to recover by following an additional step (re-updating the
+template before rolling back).
+
+Generally environment variables require far less care and attention than
+database management when it comes to maintaining zero-downtime deployments.
+However, maintainers should still understand how this process works so that they
+can handle situations where it does come up (particularly with manual rollbacks
+and the removal of environment variables).
+
+### When to update environment variables
+
+Because environment variables are updated in the task definition template, they
+must be updated _before_ the new version of the application depending on the
+variable is deployed. This includes staging, which automatically deploys on
+pushes to `main`. If a new environment variable is required and has no
+acceptable default behaviour if not configured—i.e., the application will crash
+if it is not present or the behaviour when it is not present is incompatible
+with the environment—then it must be added to the task definition template for
+the application and environment before the application is deployed. For staging,
+this means updating the task definition template before merging the PR. For
+production, this means updating the task definition template before running the
+production deployment workflow.
+
+It is generally best practice to configure acceptable default behaviours for
+undefined environment variables, but sometimes it's not possible to do so and in
+these cases it is imperative to update the task definition template before
+triggering a deployment (whether automated or manual).
+
+If the task definition template is updated _after_ the fact (whether because we
+forgot to do it beforehand or for any other reason), then the application will
+need to be redeployed so that a new task definition is created using the latest
+template revision with the updated variables.
+
+> **Note**
+>
+> _Any_ redeployment after the template is updated will receive the updated
+> variables. That means that, for example, if staging is automatically deployed
+> due to a push to `main` after updating the template, it is not necessary to
+> further redeploy staging for the application to get the updated variables. If
+> the timing of each is within seconds, however, it's best not to risk it and
+> just redeploy staging if you have any doubts.
```

Comment:
This can be changed into an [admonition](https://pradyunsg.me/furo/reference/admonitions/).